cpp/src/copying/scatter.cu

```diff
@@ -27,12 +27,14 @@
 #include <cudf/detail/utilities/cuda.cuh>
 #include <cudf/dictionary/detail/search.hpp>
 #include <cudf/lists/list_view.cuh>
+#include <cudf/scalar/scalar_factories.hpp>
 #include <cudf/strings/detail/scatter.cuh>
 #include <cudf/strings/string_view.cuh>
 #include <cudf/structs/struct_view.hpp>
 #include <cudf/table/table_device_view.cuh>
 
 #include <rmm/cuda_stream_view.hpp>
+#include "cudf/detail/iterator.cuh"
```

Comment:
This should be moved up

```suggestion
#include <cudf/detail/iterator.cuh>
```

Reply:
Ah, nice catch. I wonder if the new clang-format has some config to pick out these lines automatically?